### PR TITLE
Add basic fuzz testing suit [ECR-203]

### PR DESCRIPTION
**Pull request by vitvakatu**

### CODE REVIEW

File: exonum/fuzz/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+target
+corpus
```
Comment (ozkriff) [inline exonum/fuzz/.gitignore]: why not add these ignores to repo's `.gitignore`? So they all can be found in one place.

```text
./fuzz/target
./fuzz/corpus
./fuzz/artifacts
```

Comment (vitvakatu) [reply exonum/fuzz/.gitignore]: The only reason is that it's file was generated automatically by `cargo-fuzz`. Will fix

File: exonum/fuzz/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+cargo-fuzz = true
+
+[dependencies.exonum]
+path = ".."
+[dependencies.libfuzzer-sys]
+git = "https://github.com/rust-fuzz/libfuzzer-sys.git"
+
+# Prevent this from interfering with workspaces
+[workspace]
+members = ["."]
```
Comment (matklad) [inline exonum/fuzz/Cargo.toml]: Perhaps an `exclude` in the parent Cargo.toml would be more direct?

File: exonum/fuzz/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+authors = ["Automatically generated"]
+publish = false
+
+[package.metadata]
+cargo-fuzz = true
+
+[dependencies.exonum]
+path = ".."
+[dependencies.libfuzzer-sys]
+git = "https://github.com/rust-fuzz/libfuzzer-sys.git"
```
Comment (matklad) [inline exonum/fuzz/Cargo.toml]: Why not 

```
[dependencies]
exonum = { path = ".." }
libfuzzer-sys = { git = "..." }
```

**Comment by vitvakatu** [comment]
> Hm, should we somehow integrate it with CI?

No, it's probably not possible and it isn't the goal.

File: services/configuration/src/lib.rs [hunk 4]
```diff
@@ -255,7 +256,7 @@ where
     /// 2. Config, previous to a `StoredConfiguration` is referenced by
     /// `previous_cfg_hash` in `StoredConfiguration`.
     ///
-    /// - Table **index** is validator_id - position of a validator's `PublicKey`
+    /// - Table **index** is validator_id - position of a validators `PublicKey`
```
Comment (mvlabat) [inline services/configuration/src/lib.rs]: Singular form then?

**Pull request merged**